pkg/server/server.go

```diff
@@ -416,8 +416,27 @@ func (s *Server) Run(ctx context.Context) error {
 		s.kcpSharedInformerFactory.Start(ctx.StopCh)
 		s.rootKcpSharedInformerFactory.Start(ctx.StopCh)
 
-		s.kcpSharedInformerFactory.WaitForCacheSync(ctx.StopCh)
-		s.rootKcpSharedInformerFactory.WaitForCacheSync(ctx.StopCh)
+		if err := wait.PollImmediateInfiniteWithContext(goContext(ctx), time.Millisecond*500, func(ctx context.Context) (bool, error) {
+			startedTypes := s.kcpSharedInformerFactory.WaitForCacheSync(ctx.Done())
+			for t, started := range startedTypes {
+				if !started {
+					klog.V(3).Infof("failed to sync %s informer", t)
+					return false, nil
+				}
+			}
+			startedTypes = s.rootKcpSharedInformerFactory.WaitForCacheSync(ctx.Done())
+			for t, started := range startedTypes {
+				if !started {
+					klog.V(3).Infof("failed to sync %s informer", t)
```

Comment:
I think the default log level is `V(2)` - failing to sync an informer seems important so I'd increase the log level.